build.gradle

```diff
@@ -391,8 +391,9 @@ project('spring-integration-http') {
 	description = 'Spring Integration HTTP Support'
 	dependencies {
 		compile project(":spring-integration-core")
-		compile "org.springframework:spring-webmvc:$springVersion"
-		compile "org.springframework:spring-webflux:$springVersion"
+		compile "org.springframework:spring-web:$springVersion"
+		compile ("org.springframework:spring-webmvc:$springVersion", optional)
+		compile ("org.springframework:spring-webflux:$springVersion", optional)
```

Comment:
Well, actually I'm starting to think that we can make them all non-optional.
As I see by our tests it doesn't hurt to have MVC and WebFlux in the same CP.
That is in the end only end-user choice which engine to choose: `@EnableWebMvc` or `@EnableWebFlux` and configure particular server  - Servlet Container or Reactor.